src/compiler/sys.ts

```diff
@@ -556,6 +559,14 @@ namespace ts {
                     return process.cwd();
                 },
                 readDirectory,
+                getModifiedTime(path) {
+                    return _fs.existsSync(path) && _fs.statSync(path).mtime;
+                },
+                createHash(data) {
+                    const hash = _crypto.createHash("md5");

```

Comment:
Do we always want to use MD5 @mhegazy?

Also, could you get away with a single call to `createHash`?

```ts
if (!hash) {
    hash = _crypto.createHash("md5");
}

hash.update(data);
return hash.digest("hex");
```